src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target.hidden === false) {
+      target.health -= 50;
+    }
+
+    if (target.health <= 0) {
+      const index = Animal.alive.indexOf(target);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
```suggestion
        Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

looks like it is possible to make in one array iteration.